.ruff.toml

```
@@ -218,6 +218,9 @@ ignore = [
     "TRY300",  # Consider `else` block
     "TRY301",  # raise-within-try
 
+    # pyupgrade (UP)
+    "UP038",  # isinstance using union separators. The code is slower.
+
```

Why is this being added?

---

It only turns on with py3.10+, so we didn't detect it before.

---

I put it on the temporary list because I expect that future versions of python will fix the performance aspect of this syntax, doing some kind of in-lining. When the Union operation is as fast as a tuple then we should let Ruff fix it — `isinstance(x, this | that)` is semantically better.

---

Makes sense to me. I'd suggest noting the reason in the comment 
```suggestion
    # pyupgrade (UP)
    "UP038",  # isinstance using union separators. The code is slower as of Python 3.10-3.12

```